### PR TITLE
[libssh] Update libssh to v0.10.4

### PR DESCRIPTION
**Pull request by illera88**
**Describe the pull request**

- #### What does your PR fix?
  Updates old libssh version

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
Same ones than before

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
Yes

### CODE REVIEW

File: ports/libssh/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "libssh",
-  "version": "0.9.6",
-  "port-version": 2,
+  "version": "0.10.4",
+  "port-version": 1,
```
Comment (Adela0814) [inline ports/libssh/vcpkg.json:4]: Please remove `port-version`.

File: ports/libssh/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
   "description": "libssh is a multiplatform C library implementing the SSHv2 protocol on client and server side",
   "homepage": "https://www.libssh.org/",
   "supports": "!(uwp | arm)",
```
Comment (Adela0814) [inline ports/libssh/vcpkg.json:7]: Please add license for `libssh`, thanks.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: ports/libssh/vcpkg.json (hunk 1 above)
Comment (illera88) [reply ports/libssh/vcpkg.json]: done in latest commint

File: ports/libssh/vcpkg.json (hunk 2 above)
Comment (illera88) [reply ports/libssh/vcpkg.json:7]: done in latest commit

**Review by Adela0814 — APPROVED** [review]

**Comment by illera88** [comment]
The PR has been flagged with `requires:author-response` but I've done all the necessary changes

**Review by Adela0814 — DISMISSED** [review]

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**